libdeno/binding.cc

```diff
@@ -25,6 +25,36 @@ Deno* FromIsolate(v8::Isolate* isolate) {
   return static_cast<Deno*>(isolate->GetData(0));
 }
 
+void LazilyCreateDataMap(Deno* d) {
+  if (d->async_data_map.IsEmpty()) {
+    v8::HandleScope handle_scope(d->isolate);
+    // It's important for security reasons that async_data_map is not exposed to
+    // the VM.
+    auto async_data_map = v8::Map::New(d->isolate);
+    d->async_data_map.Reset(d->isolate, async_data_map);
+  }
+  DCHECK(!d->async_data_map.IsEmpty());
+}
+
+void AddDataRef(Deno* d, int32_t req_id, v8::Local<v8::Value> data_v) {
+  LazilyCreateDataMap(d);
+  auto async_data_map = d->async_data_map.Get(d->isolate);
+  auto context = d->context.Get(d->isolate);
+  auto req_id_v = v8::Integer::New(d->isolate, req_id);
+  auto r = async_data_map->Set(context, req_id_v, data_v);
```

Comment:
This is super duper slow. But whatever, it works, it's safe. We can always use externalize later 👿

Reply:
Ya.. we're going to have to benchmark and profile it. The other option is using `std::map<int32_t, v8::Persistent<v8::ArrayBufferView>>` which may be faster - but who knows without measuring.